pkg/controller/atlasproject/atlasproject_controller.go

```diff
@@ -379,6 +399,10 @@ func managedByAtlas(ctx context.Context, atlasClient mongodbatlas.Client) custom
 			return false, err
 		}
 
+		if project.Spec.Name == atlasProject.Name {
```

Comment:
So if the name matches we take ownership?

Reply:
We use the name to match a project if for any reason we fetch a project by its ID and they don't match the name, something is wrong and we should not proceed.